src/app/modules/payments/stripe.utils.ts

```diff
@@ -410,7 +410,21 @@ const formatProductsDescriptionHtml = (products: ProductItem[]) => {
   let final =
     '<td class="Table-description Font Font--body" style="color: #525f7f;font-size: 15px;line-height: 36px;width: 100%;">'
   products.forEach((product) => {
-    const productStr = `${product.data.name} x ${product.quantity}<br>`
+    let productStr = ''
+    // Ensure product.name is a string and product.quantity is a number
+    // todo: improve fix to throw error
+    if (!product.data || !product.data.name) {
+      logger.error({
+        message: 'Received invalid request from Stripe webhook endpoint',
+        meta: {
+          action: 'formatProductsDescriptionHtml',
+          error: Error('Invalid product data'),
+        },
+      })
+    } else {
+      const productName = encode(product.data.name)
+      productStr = `${productName} x ${product.quantity}<br>`
+    }
```

Comment:
The whole error logging seems overly defensive to me here 🤔. By the time we are at the display formatting stage (when this code runs), the data should be clean.If we want extra safety, we can just have something simple like this:

```typescript
encode(product?.data?.name || '')
```
but even that is probably not needed not be needed.